flexmeasures/cli/data_delete.py

```diff
@@ -196,4 +202,77 @@ def delete_prognoses(
     depopulate_prognoses(app.db, sensor_id)
 
 
+@fm_dev_delete_data.command("unchanged_beliefs")
+@with_appcontext
+@click.option(
+    "--sensor-id",
+    type=int,
+    help="Delete unchanged (time series) data for a single sensor only. Follow up with the sensor's ID. ",
+)
+@click.option(
+    "--delete-forecasts/--keep-forecasts",
+    "delete_unchanged_forecasts",
+    default=True,
+    help="Use the --keep-forecasts flag to keep unchanged beliefs with a positive belief horizon (forecasts).",
+)
+@click.option(
+    "--delete-measurements/--keep-measurements",
+    "delete_unchanged_measurements",
+    default=True,
+    help="Use the --keep-measurements flag to keep beliefs with a zero or negative belief horizon (measurements, nowcasts and backcasts).",
+)
+def delete_unchanged_beliefs(
+    sensor_id: Optional[int] = None,
+    delete_unchanged_forecasts: bool = False,
+    delete_unchanged_measurements: bool = False,
+):
+    """Delete unchanged beliefs (i.e. updated beliefs with a later belief time, but with the same event value)."""
+    q = db.session.query(TimedBelief)
+    if sensor_id:
+        sensor = Sensor.query.filter(Sensor.id == sensor_id).one_or_none()
+        if sensor is None:
+            print(f"Failed to delete any beliefs: no sensor found with id {sensor_id}.")
+            return
+        q = q.filter(TimedBelief.sensor_id == sensor.id)
+    num_beliefs_before = q.count()
+
+    unchanged_queries = []
+    if delete_unchanged_forecasts:
+        q_unchanged_forecasts = query_unchanged_beliefs(
+            db.session,
+            TimedBelief,
+            q.filter(
+                TimedBelief.belief_horizon > timedelta(0),
+            ),
+            include_non_positive_horizons=False,
+        )
+        unchanged_queries.append(q_unchanged_forecasts)
+    if delete_unchanged_measurements:
+        q_unchanged_measurements = query_unchanged_beliefs(
+            db.session,
+            TimedBelief,
+            q.filter(
+                TimedBelief.belief_horizon <= timedelta(0),
+            ),
+            include_positive_horizons=False,
+        )
+        unchanged_queries.append(q_unchanged_measurements)
+
+    num_beliefs_up_for_deletion = sum([q.count() for q in unchanged_queries])
+    prompt = f"Delete {num_beliefs_up_for_deletion} unchanged beliefs out of {num_beliefs_before} beliefs?"
```

Comment:
If we do treat measurements and forecasts separately, it's probably nice to also show their numbers separately.